spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Span.java

```diff
@@ -78,8 +78,8 @@ public Span(Span current, Span savedSpan) {
 		this.remote = current.isRemote();
 		this.exportable = current.isExportable();
 		this.processId = current.getProcessId();
-		this.tags.putAll(current.tags());
-		this.logs.addAll(current.logs());
+		this.tags = current.tags;
```

Comment:
I'm worried this might get un-fixed at some point.

I'd put in a test case that proves this reference is shared and/or comment why we are sharing it (ex note the use case, even if it takes two lines).